RIOT-OS/RIOT
# net: NETOPT_MAX_{PACKET_SIZE -> PDU_SIZE}

### Contribution description

This PR renames `NETOPT_MAX_PACKET_SIZE` into `NETOPT_MAX_PDU_SIZE`, which should be clearer. The documentation is updated accordingly.


### Testing procedure

Please verify that the new name and the updated documentation make sense. Apart from that Murdock will verify that this does not break anything.

### Issues/PRs references

PR https://github.com/RIOT-OS/RIOT/pull/10340 implemented `netdev_driver_t::get()` for `NETOPT_MAX_PACKET_SIZE` incorrectly and returned the maximum frame size including the layer 2 header. This rename should make clear that the maximum PDU size should be returned instead of the full frame size.

----
Update 1: Changed `NETOPT_MAX_L2_PDU_SIZE` to `NETOPT_MAX_PDU_SIZE`, as it is not only used to retrieve the layer 2 PDU.
